lib/models/debugs_bunny/concerns/has_encrypted_attributes.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'active_support/concern'
+
+module DebugsBunny
+  module HasEncryptedAttributes
+    extend ActiveSupport::Concern
+
+    included do
+      require 'attr_encrypted'
+      require 'daffy_lib'
+      include DaffyLib::PartitionProvider
+      include DaffyLib::HasEncryptedAttributes
+    end
+
+    class_methods do
+      def set_encryption_options
+        attr_encrypted_options.merge!(
+          encryptor: DaffyLib::CachingEncryptor,
+          encrypt_method: :zt_encrypt,
+          decrypt_method: :zt_decrypt,
+          partition_guid: proc { |object| object.generate_partition_guid },
+          encryption_epoch: proc { |object| object.generate_encryption_epoch },
+          cmk_key_id: proc { DebugsBunny.configuration.encryption_cmk_key_id },
+          expires_in: proc { DebugsBunny.configuration.encryption_key_cache_timeout }
+        )
+      end
+
+      def define_encrypted_attributes
+        define_singleton_method(:set_encrypted_attributes) do
+          set_encryption_options
+          yield
+        end
+        set_encrypted_attributes
```

Comment:
What's the advantage of making this a singleton method, as opposed to a regular method? Also, do you need a test for this method? Didn't see one in this PR.

Reply:
It's a singleton method since the encryption options and attributes belong to the class, and not the instance. This is used in the class definition itself:
```
class Trace < DebugsBunny::ApplicationRecord
...
define_encrypted_attributes do
      attr_encrypted :dump
end
...
end
```

Reply:
It's done this way largely to hide some of the encryption implementation details; this way, a class that includes the `HasEncryptedAttributes` module doesn't have to be concerned with the inner workings of DaffyLib

Reply:
In terms of tests, we simply test that the attributes can be read and written; if so, we know that the encryption libraries underneath are working, since there is no database column called `dump` that can be erroneously accessed with unencrypted data.

Reply:
> It's a singleton method since the encryption options and attributes belong to the class, and not the instance.

The singleton is being defined within the class method `define_encrypted_attributes`, so it seems like the functionality could have been defined directly in that class method, without the singleton. Was wondering if the singleton was preventing some issue with doing it that way. Was just curious.